presto-hive/src/test/java/com/facebook/presto/hive/statistics/TestQuickStatsProvider.java

```diff
@@ -0,0 +1,341 @@
+    private HdfsEnvironment hdfsEnvironment;
+    private DirectoryLister directoryListerMock;
+    private SemiTransactionalHiveMetastore metastoreMock;
+    private MetastoreContext metastoreContext;
+    private PartitionQuickStats mockPartitionQuickStats;
+    private PartitionStatistics expectedPartitionStats;
+
+    @BeforeTest
+    public void setUp()
+    {
+        metastoreMock = mock(SemiTransactionalHiveMetastore.class);
+        metastoreContext = new MetastoreContext(SESSION.getUser(),
+                SESSION.getQueryId(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                false,
+                HiveColumnConverterProvider.DEFAULT_COLUMN_CONVERTER_PROVIDER);
+        when(metastoreMock.getPartition(any(), eq(TEST_SCHEMA), eq(TEST_TABLE), any()))
+                .thenReturn(Optional.of(new Partition(
+                        TEST_SCHEMA,
+                        TEST_TABLE,
+                        ImmutableList.of(),
+                        new Storage(
+                                fromHiveStorageFormat(PARQUET),
+                                "some/path",
+                                Optional.empty(),
+                                true,
+                                ImmutableMap.of(),
+                                ImmutableMap.of()),
+                        ImmutableList.of(),
+                        ImmutableMap.of(),
+                        Optional.empty(),
+                        false,
+                        true,
+                        0,
+                        0)));
+
+        when(metastoreMock.getTable(any(), eq(TEST_SCHEMA), eq(TEST_TABLE)))
+                .thenReturn(Optional.of(new Table(
+                        TEST_SCHEMA,
+                        TEST_TABLE,
+                        "user_name",
+                        MANAGED_TABLE,
+                        new Storage(fromHiveStorageFormat(PARQUET),
+                                "some/location",
+                                Optional.empty(),
+                                false,
+                                ImmutableMap.of(),
+                                ImmutableMap.of()),
+                        ImmutableList.of(),
+                        ImmutableList.of(),
+                        ImmutableMap.of(),
+                        Optional.empty(),
+                        Optional.empty())));
+
+        directoryListerMock = mock(DirectoryLister.class);
+        when(directoryListerMock.list(any(), any(), any(), any(), any(), any())).thenReturn(emptyIterator());
+
+        MetastoreClientConfig metastoreClientConfig = new MetastoreClientConfig();
+        hdfsEnvironment = createTestHdfsEnvironment(hiveClientConfig, metastoreClientConfig);
+
+        ColumnQuickStats<Integer> mockIntegerColumnStats = new ColumnQuickStats<>("column", Integer.class);
```

Comment:
maybe re-use methods from `TestParquetQuickStatsBuilder`?

Reply:
Didn't get this - what methods are you referring to ?

Reply:
I meant these: https://github.com/prestodb/presto/pull/21436/files#diff-5b27abaf2f3eda821b70c980b26ea45db3bf6b495b626362ac8d350b93ff3c93R131

Am fine leaving this as is, but thought it might just be less LoC to use it rather than set each explicitly